crates/ratings_new/src/db/user.rs

```diff
@@ -16,19 +16,8 @@ pub struct User {
 }
 
 impl User {
-    /// Creates a new user from the given [`ClientHash`]
-    pub fn new(client_hash: &str) -> Self {
-        let now = OffsetDateTime::now_utc();
-        Self {
-            id: -1,
-            client_hash: client_hash.to_string(),
-            last_seen: now,
-            created: now,
-        }
-    }
-
     /// Create a [`User`] entry, or note that the user has recently been seen
-    pub async fn create_or_seen(self, conn: &mut PgConnection) -> Result<Self> {
+    pub async fn create_or_seen(client_hash: &str, conn: &mut PgConnection) -> Result<Self> {
```

Comment:
I'm curious why this is changed to accept a string reference instead of consuming the user struct

Reply:
Because the usage of this was first to construct a `User` with mostly dummy data before calling `create_or_seen` and only making use of the `client_hash` field. Changing this instead to a constructor means we don't need to stub out the dummy data for no reason and there's no change in functionality.

Reply:
That makes a lot of sense - thanks!